compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -170,19 +171,47 @@ pub(super) fn opt_const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<
                         .position(|arg| arg.id() == hir_id)
                         .map(|index| (index, seg))
                 });
+            // FIXME(associated_const_equality): recursively search through the bindings instead
```

Comment:
Not sure I understand, why do generic args not have this issue, but associated bounds do?

Reply:
to be honest I've completely forgotten why I wrote this comment. I have no idea what I meant